components/select/select.ts

```diff
@@ -87,11 +87,21 @@ export class SuiSelect<T, U> extends SuiSelectBase<T, U> {
         if (value != null) {
             if (this.options.length > 0) {
                 // If the options have already been loaded, we can immediately match the ngModel value to an option.
-                this.selectedOption = this.options.find(o => value == this.valueGetter(o));
+                this.selectedOption = this.findOption(this.options, value);
             }
             if (!this.selectedOption) {
-                // Otherwise, cache the written value for when options are set.
-                this._writtenOption = value;
+                if (this.searchService.selectedLookup) {
```

Comment:
One thing I've just thought of is potentially making `selectedLookup` a virtual property, that does the check in the getter and returns `undefined` if not.